### ISSUE TITLE
Search for pubspec.yml in parent directories

### ISSUE THREAD

**Issue by f3ath** [f3ath/pubspec-version#10]
If pubspec.yaml is not found in the current directory, search in parent directories. This behavior should only be applied for the default directory.

**Comment by Agraphie** [comment]
How many parents would it need to go up? And is this use case likely?

**Comment by f3ath** [comment]
Up to the root folder I'd say. I think it is somewhat likely. Think of Git, you can run any git command from any folder withing the repo, I find it pretty convenient.

**Comment by Agraphie** [comment]
To the root folder of the project, so not the absolute root of the file system? I'd be happy to play around with it a bit and see if I can find a solution.

**Comment by f3ath** [comment]
The filesystem root.

**Comment by Agraphie** [comment]
I have added this here https://github.com/Agraphie/pubspec-version/tree/feature/search_parent_for_pubspec

I also added tests. Do you want to create a branch here as well? I can create a pull request to that branch and we can compare and discuss better.

**Comment by f3ath** [comment]
Thanks. I'd like to see this as a separate PR without the unrelated (version manipulation) changes.

**Comment by Agraphie** [comment]
Yes true. Weird I thought I used the master branch here as HEAD. I'll reset it to exclude the version changes.
